Src/FluentAssertions/Events/EventRaisedOrder.cs

```diff
@@ -0,0 +1,20 @@
+using System.Threading;
+
+namespace FluentAssertions.Events
+{
+    /// <summary>
+    /// Tracks the order of events raised by <see cref="FluentAssertions.Events.EventMonitor{T}"/>.
+    /// </summary>
+    internal sealed class EventRaisedOrder
```

Comment:
🔧 I think the term _index_ is not entirely correct and would prefer to use the term _sequence_. And Given that this class is more of a thread-safe sequence generator, I would call it `ThreadSafeSequenceGenerator`. From that perspective, the docs are not correct either.

Reply:
Renamed in https://github.com/fluentassertions/fluentassertions/pull/1773/commits/ee0b86ba673e5491c542e9aca797095ad66db2cc. Left the `ThreadSafeSequenceGenerator` in the `FluentAssertions.Events` namespace, thought about moving to the `FluentAssertions.Common` namespace since it's a "generic" class of sorts now. Let me know what you think.

Reply:
That's the right call. Something shouldn't be generic just become it can. That leads to overuse of DRY. It is designed to be used by the event raising assertions and for nothing else.